### PR TITLE
Feature.ngsild.203

### PR DESCRIPTION
**Pull request by jicarretero**

### CODE REVIEW

File: features/203.ngsild.IoTAgentJson.feature [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+    Given I set the tutorial 203 LD
+
+  # Request 1 -
+  Scenario: Checking the IoTAGent Service health
+    When  I send GET HTTP request to "http://localhost:4041/iot/about"
+    Then  I receive a HTTP "200" response code from IoTAgent with the body "01.response.json" and exclusions "01.excludes"
+
+  # Request 2, 3 - Provision service and provision device
+  ## Modified json file -- changed /iot/d for /iot/json
+  Scenario Outline: Provisining a service Group
```
Comment (flopezag) [inline features/203.ngsild.IoTAgentJson.feature]: Provisioning...

File: features/203.ngsild.IoTAgentJson.feature [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+    And   I perform the request
+    Then  I receive a HTTP response with status 201 and empty dict
+    And   I wait "1" seconds
+    Examples:
+      | url                                | file            | description         |
+      | http://localhost:4041/iot/services | 02.request.json | Provision a service |
+      | http://localhost:4041/iot/devices  | 03.request.json | Provision a device  |
+
+  # Request 4 - Set some value to sensor
+  Scenario: Sending some simulating it from dummy iot device - Request 4
```
Comment (flopezag) [inline features/203.ngsild.IoTAgentJson.feature]: Please rephrase the Scenario, it is not clear

File: features/203.ngsild.IoTAgentJson.feature [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+    And   I perform the request
+    Then  I receive a HTTP response with status 204 and empty dict
+    Examples:
+      | file                         | device     | attr  | description            |
+      | activate.things.request.json | water001   | on    | Act. irrigation system |
+      | activate.things.request.json | tractor001 | start | Act. tractor           |
+      | activate.things.request.json | filling001 | add   | Act. filling system    |
+
+  ## ERR - 409 Conflict -- the service is previously created
+  Scenario: Req 17 - Provisining a service Group for CRUD operations
```
Comment (flopezag) [inline features/203.ngsild.IoTAgentJson.feature]: Provisioning...

File: features/203.ngsild.IoTAgentJson.feature [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+    And   I set the body request as described in 25.request.json
+    And   I perform the request
+    Then  I receive a HTTP response with status 200 and empty dict
+
+  Scenario: Req 26 - Delete a provisioned device
+    When  I prepare a DELETE HTTP request to "http://localhost:4041/iot/devices/water002"
+    And   I set header fiware-service to openiot
+    And   I set header fiware-servicepath to /
+    And   I perform the query request
+    Then  I receive a HTTP response with status 204 and empty dict
```
Comment (flopezag) [inline features/203.ngsild.IoTAgentJson.feature]: Add new line

File: features/203.ngsild.IoTAgentJson.feature [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+
+  ## ERR - Modified c.value in response.json from "1" to 1 (str to int)
+  ## ERR - No "@context" in response => failure
+  Scenario: Req 7 - Test the value of the new device
+    When  I prepare a GET HTTP request to "http://localhost:1026/ngsi-ld/v1/entities/?type=Device"
+    And   I set header NGSILD-Tenant to openiot
+    And   I set header NGSILD-Path to /
+    And   I set header Link to <http://context/ngsi-context.jsonld>; rel="http://www.w3.org/ns/json-ld#context"; type="application/ld+json"
+    And   I perform the query request
+    And   I filter the result with jq .[]|select(.id == "urn:ngsi-ld:Device:motion003")
```
Comment (flopezag) [inline features/203.ngsild.IoTAgentJson.feature]: For the clarity of the step definition, I would suggest the following description:

I filter the result with jq '.[]|select(.id == "urn:ngsi-ld:Device:motion003")'

File: features/steps/203.ngsild.IotAgentJson.py [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+from os.path import join
+import jq
+
+
+@given(u'I set the tutorial 203 LD')
+def step_impl_tutorial_203(context):
+    context.data_home = join(join(join(CODE_HOME, "features"), "data"),  "203.ld.IotAgentJSON")
+
+
+@step(u'I filter the result with jq {expr}')
```
Comment (flopezag) [inline features/steps/203.ngsild.IotAgentJson.py]: If we take into consideration the comment of the .feature the step should be defined like:

@step(u"I filter the result with jq '{expr}'")

**Review by flopezag — CHANGES_REQUESTED** [review]

File: features/203.ngsild.IoTAgentJson.feature [hunk 7]
```diff
@@ -0,0 +1,213 @@
+Feature: test tutorial 201.Introduction to IoT Sensors
```
Comment (flopezag) [inline features/203.ngsild.IoTAgentJson.feature]: Please, do not forget this line...

**Review by flopezag — CHANGES_REQUESTED** [review]

File: features/203.ngsild.IoTAgentJson.feature (hunk 7 above)
Comment (jicarretero) [reply features/203.ngsild.IoTAgentJson.feature]: Agree. Changes done

File: features/steps/203.ngsild.IotAgentJson.py (hunk 6 above)
Comment (jicarretero) [reply features/steps/203.ngsild.IotAgentJson.py]: Ok. Done

**Review by flopezag — APPROVED** [review]

**Pull request merged**